### PR TITLE
Wifi components

### PR DESCRIPTION
**Pull request by ChinaTizenRT**
@xixidodo 
@xiarihf  
@zhouxinhe 
@SRCN-LiQin

### CODE REVIEW

File: external/Kconfig [hunk 1]
```diff
@@ -20,3 +20,4 @@ source "$EXTERNALDIR/WICED/Kconfig"
 source "$EXTERNALDIR/libopus/Kconfig"
 source "$EXTERNALDIR/stress_tool/Kconfig"
 source "$EXTERNALDIR/swepd/Kconfig"
+source "$EXTERNALDIR/esp_idf_port/Kconfig"
```
Comment (SRCN-LiQin) [inline external/Kconfig:23]: as compre to other directory in external, like WICED,  is it a proper directory name ?

**Comment by ChinaTizenRT** [comment]
this PR only include wifi lib, no other components for wifi feature.

**Comment by xiarihf** [comment]
Can you add libs version information? Such as the commit No. in IDF, or other comment.

**Review by xixidodo — COMMENTED** [review]
1. 
Please check commit rule https://github.com/Samsung/TizenRT/wiki/Documentations#commit-rules
then modify the commit message according to that

2. add more in PR description also

**Comment by xixidodo** [comment]
> Can you add libs version information? Such as the commit No. in IDF, or other comment.

Agreed.  Maybe a  README file in 'esp_idf_port' to explictly describle idf commit hash and other needed comments

**Pull request merged**